File: Kaviyarasan/Apirl-26/Art-Gallery/fix.js
const fs = require('fs');
const glob = require('fs').readdirSync('pages').filter(f => f.endsWith('.html')).map(f => 'pages/' + f);
glob.push('index.html');

glob.forEach(file => {
    if (!fs.existsSync(file)) return;
    let content = fs.readFileSync(file, 'utf8');

    // First clean up PREVIOUS mess
    const badDesktopRegex = /\n?\s*<div class="navbar__dropdown">\s*<a href="#" class="navbar__link navbar__dropdown-trigger" aria-expanded="false">Dashboard <svg[^>]+><polyline[^>]+><\/svg><\/a>\s*<div class="navbar__dropdown-menu" role="menu">\s*<a href="dashboard\.html" class="navbar__dropdown-item" role="menuitem">User Dashboard<\/a>\s*<a href="dashboard\.html#admin" class="navbar__dropdown-item" role="menuitem">Admin Dashboard<\/a>\s*<\/div>\s*<\/div>/g;
    
    content = content.replace(badDesktopRegex, '');
    
    const badMobileRegex = /\s*<a href="dashboard\.html" class="navbar__link" style="visibility:visible;opacity:1;transform:none;">User Dashboard<\/a>\s*<a href="dashboard\.html#admin" class="navbar__link" style="visibility:visible;opacity:1;transform:none;">Admin Dashboard<\/a>/g;
    
    content = content.replace(badMobileRegex, '');

    // NOW INSERT CORRECTLY
    const desktopDropdown = `        <div class="navbar__dropdown">
          <a href="#" class="navbar__link navbar__dropdown-trigger" aria-expanded="false">Dashboard <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2"><polyline points="6 9 12 15 18 9"/></svg></a>
          <div class="navbar__dropdown-menu" role="menu">
            <a href="dashboard.html" class="navbar__dropdown-item" role="menuitem">User Dashboard</a>
            <a href="dashboard.html#admin" class="navbar__dropdown-item" role="menuitem">Admin Dashboard</a>
          </div>
        </div>
      </div>`;

    const mobileLinks = `      <a href="dashboard.html" class="navbar__link" style="visibility:visible;opacity:1;transform:none;">User Dashboard</a>
      <a href="dashboard.html#admin" class="navbar__link" style="visibility:visible;opacity:1;transform:none;">Admin Dashboard</a>
      <div class="navbar__mobile-toggles" style="display:flex;gap:12px;margin-top:16px;">`;

    // 1. Insert desktop: Look for `navbar__links` closing div. We can split.
    if (content.includes('navbar__links') && !content.includes('User Dashboard')) {
       // Replace the </div> that closes navbar__links
       // We know the last link before </div> is contact
       content = content.replace(/(<a href="contact\.html" class="navbar__link[^>]*>Contact<\/a>\s*)<\/div>/g, '$1' + desktopDropdown);
       
       // 2. Insert mobile: we want it before the toggles in mobile menu, or before Login button.
       content = content.replace(/(<div style="display:flex;gap:12px;margin-top:16px;">)/g, mobileLinks);
       
        // Some pages don't have toggler, just login button in mobile
        content = content.replace(/(<a href="login\.html" class="btn btn--primary"[^>]*>Login<\/a>\s*<\/div>\s*<\/nav>)/g, '      <a href="dashboard.html" class="navbar__link">User Dashboard</a>\n      <a href="dashboard.html#admin" class="navbar__link">Admin Dashboard</a>\n$1');
    }
    
    fs.writeFileSync(file, content, 'utf8');
});
console.log('Cleaned and Fixed');

File: Kaviyarasan/Apirl-26/Art Gallery/fix.py
import os
import glob
import re

html_files = glob.glob('pages/*.html') + ['index.html']
css_file = 'assets/css/style.css'

# Update CSS
with open(css_file, 'r', encoding='utf-8') as f:
    css_content = f.read()

css_content = css_content.replace('animation: kenBurns 20s ease infinite alternate;', '/* animation removed */')
css_content = css_content.replace('.hero-slider__slide.active { opacity: 1; position: relative; }', '.hero-slider__slide.active { opacity: 1; z-index: 1; position: absolute; }')
css_content = css_content.replace('.hero-slider { position: relative; }', '.hero-slider { position: relative; min-height: 100vh; }')

if '.h-scroll' not in css_content:
    css_content += '''
/* --- HORIZONTAL SCROLL FIX --- */
.h-scroll {
  display: flex;
  overflow-x: auto;
  gap: var(--space-4);
  padding-bottom: var(--space-2);
  -webkit-overflow-scrolling: touch;
  scroll-snap-type: x mandatory;
}
.h-scroll > * {
  scroll-snap-align: start;
  flex-shrink: 0;
}
@media (max-width: 768px) {
  .section { padding-block: 3rem; }
  .grid { gap: var(--space-3); }
  .navbar__inner { padding-inline: var(--space-2); }
  .container { padding-inline: var(--space-2); overflow-x: hidden; }
}
'''
with open(css_file, 'w', encoding='utf-8') as f:
    f.write(css_content)

# Regex to find contact link and insert dashboard dropdown
desktop_repl = r'\1\n        <div class="navbar__dropdown"><a href="#" class="navbar__link navbar__dropdown-trigger" aria-expanded="false">Dashboard <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2"><polyline points="6 9 12 15 18 9"/></svg></a><div class="navbar__dropdown-menu" role="menu"><a href="dashboard.html" class="navbar__dropdown-item" role="menuitem">User Dashboard</a><a href="dashboard.html#admin" class="navbar__dropdown-item" role="menuitem">Admin Dashboard</a></div></div>'
mobile_repl = r'<a href="dashboard.html" class="navbar__link">User Dashboard</a>\n      <a href="dashboard.html#admin" class="navbar__link">Admin Dashboard</a>\n      \1'

for file in html_files:
    if not os.path.isfile(file):
        continue
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    # Remove inline kenBurns
    content = content.replace('style="animation:kenBurns 25s ease infinite alternate;"', '')

    # Insert Desktop Dashboard Dropdown after Contact
    if 'Admin Dashboard' not in content and 'navbar__links' in content:
        content = re.sub(r'(<a href="contact\.html" class="navbar__link[^"]*">Contact</a>)', desktop_repl, content)

    # Insert Mobile Dashboard Dropdown before <div style="display:flex;gap:12px;margin-top:16px;">
    if 'Admin Dashboard' not in content and 'navbar__mobile-menu' in content:
        content = re.sub(r'(<div style="display:flex;gap:12px;margin-top:16px;">)', mobile_repl, content)

    # Also for auth pages (no theme/rtl toggler, but has login button)
    if 'Admin Dashboard' not in content and 'navbar__mobile-menu' in content:
        content = re.sub(r'(<a href="login\.html" class="btn btn--primary"[^>]*>Login</a>)', mobile_repl, content)

    with open(file, 'w', encoding='utf-8') as f:
        f.write(content)
print("Done Phase 3")
